### ISSUE TITLE
1.5 Hours Course

### ISSUE THREAD

**Issue by anish-kum** [anish-kum/PPP-Spark-2025-2026#9]
The title says 90-minute PPP, but it was longer by a bit.

**Comment by anish-kum** [comment]
<img width="1279" height="552" alt="Image" src="https://github.com/user-attachments/assets/b16c083c-a4e7-4a5f-925c-102b4b4e4a44" />

<img width="1273" height="546" alt="Image" src="https://github.com/user-attachments/assets/eb13424c-72a8-4158-bf3e-d310ba657864" />

<img width="1279" height="548" alt="Image" src="https://github.com/user-attachments/assets/c49aa60f-5bb8-4107-b202-c741bf983e51" />

What I was learning:

- Activation Functions: Functions that introduce non-linearity into neural networks, allowing them to learn complex patterns.

- Sigmoid Function: Outputs values between 0 and 1, useful for binary classification but can suffer from vanishing gradients.

- Tanh Function: Outputs values between -1 and 1, often preferred over sigmoid due to its zero-centered output.

- ReLU (Rectified Linear Unit): Outputs the input directly if positive; otherwise, it outputs zero, helping to mitigate vanishing gradient issues.

- Leaky ReLU: A variation of ReLU that allows a small, non-zero gradient when the input is negative, preventing dead neurons.

- Softmax Function: Converts logits into probabilities for multi-class classification, ensuring the outputs sum to one.

- Choosing Activation Functions: The choice of activation function can significantly impact model performance and convergence speed.

Questions the content didn't cover that I'd like an answer to: 

- Why does the choice of an activation function matter for how well a neural network learns patterns?

- How could using different activation functions change the kinds of problems a neural network is good at solving?

Note: Finally finished the theory part of the course which was very lengthy. I know I said last week I would research in preparation to pivot to a new project (Discord bots or something else) but decided to go on with the course since this first section was close to being done. I'll probably start researching this week or next week for a new project and start since it's true I've lost some interest in this particular topic partly due to a very high learning curve as the course progresses. I'll probably come back to this later once I start a new project.

**Comment by drewmca-pkwy** [comment]
@anish-kum Anish, this looks good. Great to see both the questions as well as your commentary at the end. Pushing to the close of the unit makes sense, so long as you maintain motivation to do so. Looking forward to the next iteration.
